### PR TITLE
Add pkce with cas authentication strategy

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: src/main/scala/com/linagora/openpaas/gatling/core/authentication/pkceWithCas/PKCEWithCasSteps.scala [hunk 1]
```diff
@@ -144,4 +154,66 @@ object PKCEWithCasSteps {
...
+
+  private def logoutCasLandingPage = {
+    http("logout CAS landing page")
+      .get(CasBaseUrl + "/logout")
+      .headers(Map(
+        "Accept" -> "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+        "upgrade-insecure-requests" -> "1"
+      ))
+      .formParam("SAMLRequest", "${cas_logout_saml_request}")
+      .disableFollowRedirect
```
Comment (rouazana) [inline src/main/scala/com/linagora/openpaas/gatling/core/authentication/pkceWithCas/PKCEWithCasSteps.scala:173]: do you need this if the return code is 200?

File: src/main/scala/com/linagora/openpaas/gatling/core/authentication/pkceWithCas/PKCEWithCasSteps.scala [hunk 2]
```diff
@@ -144,4 +154,66 @@ object PKCEWithCasSteps {
...
+  private def logoutConfirmation = {
+    http("Logout confirmation")
+      .post(LemonLDAPPortalUrl + "/oauth2/logout")
+      .headers(Map(
+        "Accept" -> "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+        "upgrade-insecure-requests" -> "1",
+        "Content-Type" -> "application/x-www-form-urlencoded"))
+      .queryParam("id_token_hint", "${id_token}")
+      .queryParam("post_logout_redirect_uri", s"${OpenPaaSBaseUrl}/${InboxSpaPath}")
+      .disableFollowRedirect
```
Comment (rouazana) [inline src/main/scala/com/linagora/openpaas/gatling/core/authentication/pkceWithCas/PKCEWithCasSteps.scala:200]: idem

File: src/main/scala/com/linagora/openpaas/gatling/core/authentication/pkceWithCas/PKCEWithCasSteps.scala [hunk 3]
```diff
@@ -144,4 +154,66 @@ object PKCEWithCasSteps {
...
+
+  private def logoutGoToConfirmationPage = {
+    http("Logout go to confirmation page")
+      .get(LemonLDAPPortalUrl + "/oauth2/logout")
+      .queryParam("id_token_hint", "${id_token}")
+      .headers(Map(
+        "Accept" -> "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+        "upgrade-insecure-requests" -> "1"))
+      .queryParam("post_logout_redirect_uri", s"${OpenPaaSBaseUrl}/${InboxSpaPath}")
+      .disableFollowRedirect
```
Comment (rouazana) [inline src/main/scala/com/linagora/openpaas/gatling/core/authentication/pkceWithCas/PKCEWithCasSteps.scala:216]: idem

**Review by rouazana — APPROVED** [review]
can you complete the README?

**Review by rouazana — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]

**Review by chibenwa — COMMENTED** [review]
:+1:

**Pull request merged**